django/db/models/fields/__init__.py

```diff
@@ -828,18 +828,17 @@ def get_flatchoices(self, include_blank=True,
         first_choice = blank_choice if include_blank else []
         return first_choice + list(self.flatchoices)
 
+    @warn_about_renamed_method('Field', '_get_val_from_obj', 'value_from_object',
+                               RemovedInDjango21Warning)
     def _get_val_from_obj(self, obj):
-        if obj is not None:
```

Comment:
Here's a regression test that passes on master but crashes with the new version (after the deprecation warning is silenced):
```diff
diff --git a/tests/model_fields/tests.py b/tests/model_fields/tests.py
index 4677962..b2845ff 100644
--- a/tests/model_fields/tests.py
+++ b/tests/model_fields/tests.py
@@ -31,6 +31,11 @@ from .models import (
 
 
 class BasicFieldTests(test.TestCase):
+
+    def test_get_val_from_obj(self):
+        f = BooleanModel()
+        self.assertEqual(f._meta.get_field('string')._get_val_from_obj(None), 'abc')
+
     def test_show_hidden_initial(self):
         """
         Regression test for #12913. Make sure fields with choices respectdiff --git a/tests/model_fields/tests.py
```